base/Process.php

```diff
@@ -99,19 +99,16 @@ public function isRunning(): bool {
   }
 
   public function stop(): void {
-    if (!$this->isRunning()) {
-      return;
-    }
-
     if (is_resource($this->stdin)) {
       pclose($this->stdin);
+      $this->stdin = null;
     }
     if (is_resource($this->stdout)) {
       pclose($this->stdout);
+      $this->stdout = null;
     }
-    $pid = $this->getPid();
-    if ($pid !== null) {
-      posix_kill($pid, SIGTERM);
+    if (is_resource($this->process)) {
+      proc_terminate($this->process);
```

Comment:
proc_terminate() is consistent - the problem is that what $this->process refers to isn't.

Bash just execs() in simple cases, so $this->process is nginx/hhvm/siege/whatever. Dash *always* forks and execs, so $this->process is referring to the shell. This gets us weirdness.